evm/consensus/pow.py

```diff
@@ -0,0 +1,36 @@
+from pyethash import mkcache_bytes, hashimoto_light
+
+from evm.utils.hexidecimal import (
+    encode_hex,
+)
+from evm.exceptions import (
+    ValidationError,
+)
+from evm.utils.numeric import (
+    big_endian_to_int,
+)
+
+
+def check_pow(block_number, mining_hash, mix_hash, nonce, difficulty):
+    if len(mix_hash) != 32:
+        raise ValidationError(
+            "mix_hash should have length 32, got {0}: {1}".format(
```

Comment:
You can use `evm.validation.validate_length` for this.  The error message won't be quite as specific but I think that's ok for now.